delivery_carrier_label_gls/stock.py

```diff
@@ -0,0 +1,337 @@
+        shipping_date = datetime.strptime(
+            shipping_date, DEFAULT_SERVER_DATETIME_FORMAT)
+        delivery = {}
+        delivery.update({
+            'consignee_ref': picking.name[:20],
+            'additional_ref_1': u'',
+            'additional_ref_2': picking.name[:20],
+            'shipping_date': shipping_date.strftime('%Y%m%d'),
+            'commentary': picking.note,
+            'parcel_total_number': number_of_packages,
+        })
+        return delivery
+
+    def _prepare_sender_gls(self, cr, uid, pick, context=None):
+        partner = self.pool['stock.picking.out']._get_label_sender_address(
+            cr, uid, pick, context=context)
+        global_infos = self._prepare_global_gls(cr, uid, context=context)
+        sender = {'contact_id': pick.company_id.gls_fr_contact_id,
+                  'customer_id': global_infos['carrier_gls_customer_code'],
+                  'contact_id_inter': pick.company_id.gls_inter_contact_id,
+                  'outbound_depot': global_infos['carrier_gls_warehouse']}
+        if partner.country_id:
+            sender['country'] = partner.country_id.name
+        sender.update({
+            'shipper_street': partner.street,
+            'shipper_street2': partner.street2,
+            'shipper_name': partner.name,
+            'shipper_country': partner.country_id.code,
+            'shipper_zip': partner.zip,
+            'shipper_city': partner.city,
+        })
+        return sender
+
+    def _prepare_pack_gls(
+            self, cr, uid, tracking, weight=None, context=None):
+        global PACK_NUMBER
+        pack = {}
+        PACK_NUMBER += 1
+        pack.update({
+            'parcel_number_label': PACK_NUMBER,
+            'parcel_number_barcode': PACK_NUMBER,
+            'custom_sequence': self._get_sequence(
+                cr, uid, 'gls', context=context),
+        })
+        if weight:
+            pack.update({
+                'weight': "{0:05.2f}".format(weight),
+            })
+        else:
+            if tracking.move_ids:
+                tracking_weight = [move.weight
+                                   for move in tracking.move_ids][0]
+                pack.update({
+                    'weight': "{0:05.2f}".format(tracking_weight),
+                })
+        return pack
+
+    def _get_tracking_ids_from_moves(self, cr, uid, picking, context=None):
+        """ get all the trackings of the picking
+            no tracking_id will return a False (Browse Null), meaning that
+            we want a label for the picking
+        """
+        return sorted(set(
+            line.tracking_id for line in picking.move_lines
+        ), key=attrgetter('name'))
+
+    def _get_weight_from_moves_without_tracking(
+            self, cr, uid, picking, context=None):
+        weights = [line.weight for line in picking.move_lines
+                   if not line.tracking_id]
+        return sum(weights)
+
+    def _generate_gls_label(
```

Comment:
As we generate multiple labels I would prefer to call this method _generate_gls_labels